analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -176,6 +175,51 @@ public void Execute_PersistSymbols_InsideBlock()
                 ((ILocalReferenceOperation)operation.Instance).Local;
         }
 
+        [TestMethod]
+        public void Execute_UnusedVariable_ClearedAfterBlock()
+        {
+            const string code = @"
+var first = boolParameter ? true : false;
+Tag(""BeforeLastUse""); 
+bool second = first;
+if(boolParameter)
+    boolParameter.ToString();
+Tag(""AfterLastUse"");
+";
+            ISymbol firstSymbol = null;
+            var postProcess = new PostProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.TrackedSymbol() is { } symbol)
+                {
+                    if (symbol.Name == "first")
+                    {
+                        firstSymbol = symbol;
+                    }
+                }
+                return x.State;
+            });
+            var validator = SETestContext.CreateCS(code, postProcess).Validator;
+            validator.TagStates("BeforeLastUse").Should().HaveCount(2).And.OnlyContain(x => x[firstSymbol] != null);
+            validator.TagStates("AfterLastUse").Should().HaveCount(1).And.OnlyContain(x => x[firstSymbol] == null);
+        }
+
+        [TestMethod]
+        public void Execute_CapturedVariable_NotCleared()
+        {
+            const string code = @"
+void LocalFunction()
+{
+boolParameter = false;
+var misc = true;
+  if(misc)
+    misc.ToString();
+Tag(""LocalFunctionEnd"", boolParameter);
```

Comment:
The `Tag` will prevent cleanup from happening and the assertion assumes it's protected by it.
```suggestion
Tag(""LocalFunctionEnd"");
```

Reply:
The assertion needs to be updated accordingly. Currently, we assert that there is exactly one state - that could be an empty state when all symbols are cleared by LVA.
We should assert something like `.And.OnlyContains(x => x.SymbolsWith(BoolConstraint.True).Length == 1)` or something like that